src/commands/createFunction/FunctionListStep.ts

```diff
@@ -47,10 +47,9 @@ export class FunctionListStep extends AzureWizardPromptStep<IFunctionWizardConte
             const templates: IFunctionTemplate[] = await ext.templateProvider.getFunctionTemplates(context, context.projectPath, language, version, TemplateFilter.All);
             const foundTemplate: IFunctionTemplate | undefined = templates.find((t: IFunctionTemplate) => {
                 if (options.templateId) {
-                    const actualId: string = t.id.toLowerCase();
-                    const expectedId: string = options.templateId.toLowerCase();
-                    // Check for a match with or without the language in the id
-                    return actualId === expectedId || actualId === `${expectedId}-${language.toLowerCase()}`;
+                    const actualName: string = t.name.toLowerCase().replace(/\s/, '');
```

Comment:
We can't use the name 😬. It's a display name that could change or could be localized. We've had a bug in the past related to this, especially because the templates are returned from the feed and could change without us testing. The id is much more reliable and shouldn't change. Also, it only really comes in two formats that you need to handle:
- HttpTrigger-JavaScript
- Azure.Function.CSharp.HttpTrigger.3.x

You should be able to figure out some conditions or regex magic to handle those two cases. Your "includes" logic before was close, but it just needs to be more careful about finding the whole word without extra stuff at the end. There's probably some regex magic you could do

Reply:
Oh yeah, I remember when that broke us a while ago... 

Anyway, I think using the boundary characters should work to match _only_ whole word cases without any extra alphanumeric characters.